doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -0,0 +1,183 @@
+As soon as the update finishes succesfully, the recovery flag is cleared and the device proceeds with normal operation.
+
+Default recovery firmware
+*************************
+
+Nordic provides a default recovery firmware that can be used in the recovery process.
+This firmware uses Bluetooth LE and SMP as a transport.
+It is optimized for memory usage, currently using around 164 kB of MRAM (72 kB of application core and 92 kB radio core).
+
+.. note::
+   The default recovery firmware does not support updating from external flash memory.
+
+To use the firmware:
+
+1. Create :file:`recovery.overlay` and :file:`recovery_hci_ipc.ovelay` files in the main application's :ref:`configuration_system_overview_sysbuild` directory.
+   These files must define the ``cpuapp_recovery_partition`` and ``cpurad_recovery_partition`` nodes respectively.
+   These partitions specify where the images for the recovery firmware are stored and cannot overlap with the main application partitions.
+   For reference, see the files in the ``samples/suit/smp_transfer`` sample.
+
+#. Set the ``SB_CONFIG_SUIT_BUILD_RECOVERY`` sysbuild configuration option in the main application.
+   This will cause the recovery firmware to be built automatically as part of the main application build.
+
+#. Flash the main application firmware to the device:
+
+      .. code-block:: bash
+
+         west flash
+
+   This will automatically flash both the main application and the recovery firmware to the device.
+
+To perform the recovery perform an update in the same way as described in :ref:`nrf54h_suit_sample`.
+
+Further information about the default recovery firmware can be found in :ref:`suit_recovery`.
+The code for the default recovery firmware can be found in the :file:`samples/suit/recovery` directory.
+
+Updating the recovery firmware
+******************************
+
+To update the recovery firmware you can either use:
+
+* The APP_RECOVERY envelope, found in :file:`<main_application_build_directory>/build/DFU/app_recovery.suit`
+* The zip file, found in :file:`<main_application_build_directory>/build/zephyr/dfu_suit_recovery.zip``
+
+These can be used to update the recovery application the same as :file:`root.suit` or :file:`dfu_suit.zip` are used to update the main application - see :ref:`nrf54h_suit_sample` as an example.
+
+.. note::
+   The recovery application can only be updated from the main application - not when running the recovery application itself.
+
+Create custom recovery images
+*****************************
+
+To turn an application into a recovery application, the following steps have to be performed:
+
+1. For each of the recovery firmware images, ensure the following configuration is present:
+
+      .. code-block:: cfg
+
+         CONFIG_SUIT_RECOVERY=y
+         CONFIG_SUIT_MPI_GENERATE=n
+         CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE=n
+         CONFIG_NRF_REGTOOL_GENERATE_UICR=n
+         CONFIG_NRF_REGTOOL_GENERATE_BICR=n
```

Comment:
Don't use a code-block, but a bullet list with each option linked (`:kconfig:option:`).

Reply:
I tried to do it, but there are several problems with that approach - there is no easy way to show which version is set to `y` or `n` and it does not give the possibility to simply copy and paste the configuration into the Kconfig file

Reply:
Being able to copy-paste code snippets is a nice-have for the documentation, not a must. Completeness and consistency is a must. Our guidelines state how to link Kconfig options - https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/documentation/styleguide.html#kconfig - and that's what we are striving to do elsewhere in the docs. You don't link these Kconfigs earlier in the doc, so they need to be linked here.

Reply:
I'll take down my request changes, but please discuss this with @FrancescoSer .

Reply:
Ok, changed as we've discussed